source/common/tcp_proxy/upstream.cc

```diff
@@ -287,10 +287,10 @@ void Http2Upstream::setRequestEncoder(Http::RequestEncoder& request_encoder, boo
   });
 
   if (config_.usePost()) {
-    headers->addReference(Http::Headers::get().Path, "/");
+    headers->addReference(Http::Headers::get().Path, config_.postPath());
     headers->addReference(Http::Headers::get().Scheme, scheme);
   } else if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.use_rfc_connect")) {
-    headers->addReference(Http::Headers::get().Path, "/");
+    headers->addReference(Http::Headers::get().Path, config_.postPath());
```

Comment:
I don't think you should set path for CONNECT?

Reply:
It needs to force extended CONNECT https://www.rfc-editor.org/rfc/rfc8441, i.e. set :protocol.

Reply:
got it, I can allow the path when use_rfc_connect is true.

Reply:
emm... this is deprecated, we said this is deprecated extended CONNECT rfc, I'm confused https://github.com/envoyproxy/envoy/blob/b3456877cdcf418294b2b3b3db21487e0c425059/changelogs/1.23.0.yaml#L152

Reply:
There is a long conversation about this https://github.com/envoyproxy/envoy/issues/20378, I think it means this is a wrong usage of extended CONNECT. @kyessenov, so we needn't support custom path for it, right?

Reply:
No, I just mean that setting path must be coupled with extended connect. You shouldn't set path for regular connect.

Reply:
oh, got it, will update, thanks